activerecord/test/cases/primary_keys_test.rb

```diff
@@ -287,7 +287,7 @@ def test_set_primary_key_with_no_connection
       assert NoConnection.connected?
       assert_equal "foo", NoConnection.primary_key
 
-      ActiveRecord::Base.connection_handler.remove_connection_pool("PrimaryKeyWithNoConnectionTest::NoConnection")
+      NoConnection.remove_connection_pool
```

Comment:
I updated this test in https://github.com/rails/rails/pull/48473 which is how I discovered the issue with `remove_connection`. I had to use the connection handler directly in order to remove the pool because `remove_connection` was resulting in `NoConnection` falling back to `ActiveRecord::Base` when it should have been a unique connection.